src/helpers/storage.js

```diff
@@ -1,34 +1,135 @@
   const newTaskList = currentTasks.filter((task) => task !== currentTasks[indexOfDeletedTask]);
-  saveDataInStorageForKey(STORAGE_KEY, newTaskList);
+  saveDataInStorageForKey(STORAGE_TASK_KEY, newTaskList);
 };
 
 export const updateTaskInStorage = (task) => {
   const currentTasks = getTasksDataFromLocalStorage();
-  const updatedTasks = currentTasks.map((currentTask) => {
-    if (currentTask.id === task.id) {
-      return task;
+  const updatedTasks = currentTasks.map((currentTask) =>
+    currentTask.id === task.id ? task : currentTask,
+  );
+  saveDataInStorageForKey(STORAGE_TASK_KEY, updatedTasks);
+};
+
+export const getGroupOfTasksDataFromLocalStorage = () => {
+  return JSON.parse(localStorage.getItem(STORAGE_GROUP_OF_TASKS_KEY)) || [];
+};
+
+// THIS IS ANOTHER SOLUTION OF updateTaskInGroupStorage LEFT BY PURPOSE
+
+// export const updateTaskInGroupStorage = (task) => {
+//   const currentGroupOfTasks = getGroupOfTasksDataFromLocalStorage();
+//   console.log(currentGroupOfTasks, task);
+//   const foundTaskGroup = currentGroupOfTasks.find((group) => {
+//     return group.tasks.find((item) => item.id === task.id);
+//   });
+//   const newUpdatedTasksInGroup = foundTaskGroup.tasks.map((item) =>
+//     item.id === task.id ? task : item,
+//   );
+//   const newUpdatedGroup = {
+//     ...foundTaskGroup,
+//     tasks: newUpdatedTasksInGroup,
+//   };
+
+//   const updatedAllGroups = currentGroupOfTasks.map((groupItem) =>
+//     groupItem.id === newUpdatedGroup.id ? newUpdatedGroup : groupItem,
+//   );
+//   saveDataInStorageForKey(STORAGE_GROUP_OF_TASKS_KEY, updatedAllGroups);
+// };
+
+export const updateTaskInGroupStorage = (task) => {
+  const currentGroupOfTasks = getGroupOfTasksDataFromLocalStorage();
+  const updatedAllGroups = [...currentGroupOfTasks].map((group) => {
+    if (group.id === task.groupId) {
+      return {
+        ...group,
+        tasks: group.tasks.map((taskItem) => (taskItem.id === task.id ? task : taskItem)),
+      };
+    }
+    return group;
+  });
+
+  saveDataInStorageForKey(STORAGE_GROUP_OF_TASKS_KEY, updatedAllGroups);
+};
+
+export const deleteTaskInGroupStorage = (task, id) => {
+  const currentGroupOfTasks = getGroupOfTasksDataFromLocalStorage();
+  const foundGroupOfTasks = [...currentGroupOfTasks].find((group) => group.id === task.groupId);
+  const indexOfDeletedTask = foundGroupOfTasks.tasks.findIndex((tasks) => tasks.id === id);
+  const newListOfTasksInFoundGroupOfTasks = foundGroupOfTasks.tasks.filter(
+    (tasksList) => tasksList !== foundGroupOfTasks.tasks[indexOfDeletedTask],
+  );
+  foundGroupOfTasks.tasks = newListOfTasksInFoundGroupOfTasks;
+  const updatedGroupsOfTasks = currentGroupOfTasks.map((group) =>
+    group.id === foundGroupOfTasks.id ? foundGroupOfTasks : group,
+  );
+
+  saveDataInStorageForKey(STORAGE_GROUP_OF_TASKS_KEY, updatedGroupsOfTasks);
+};
+
+// THIS IS ANOTHER SOLUTION OF deleteTaskInGroupStorage LEFT BY PURPOSE
```

Comment:
In code base we do not leave that kind of comments - please remove, before that you can create another branch which will not be merged, but you could still switch and get back to this if you will need

Reply:
done